pkg/cmd/serve.go

```diff
@@ -50,116 +53,142 @@ func RegisterServeFlags(cmd *cobra.Command, config *server.Config) error {
+	apiFlags.BoolVar(&config.DisableVersionResponse, "disable-version-response", false, "disables version response support in the API")
+	apiFlags.Uint16Var(&config.MaximumUpdatesPerWrite, "write-relationships-max-updates-per-call", 1000, "maximum number of updates allowed for WriteRelationships calls")
+	apiFlags.IntVar(&config.MaxCaveatContextSize, "max-caveat-context-size", 4096, "maximum allowed size of request caveat context in bytes. A value of zero or less means no limit")
+	apiFlags.IntVar(&config.MaxRelationshipContextSize, "max-relationship-context-size", 25000, "maximum allowed size of the context to be stored in a relationship")
+	apiFlags.DurationVar(&config.StreamingAPITimeout, "streaming-api-response-delay-timeout", 30*time.Second, "max duration time elapsed between messages sent by the server-side to the client (responses) before the stream times out")
+	apiFlags.DurationVar(&config.WatchHeartbeat, "watch-api-heartbeat", 1*time.Second, "heartbeat time on the watch in the API. 0 means to default to the datastore's minimum.")
+	apiFlags.Uint32Var(&config.MaxReadRelationshipsLimit, "max-read-relationships-limit", 1000, "maximum number of relationships that can be read in a single request")
+	apiFlags.Uint32Var(&config.MaxDeleteRelationshipsLimit, "max-delete-relationships-limit", 1000, "maximum number of relationships that can be deleted in a single request")
+	apiFlags.Uint32Var(&config.MaxLookupResourcesLimit, "max-lookup-resources-limit", 1000, "maximum number of resources that can be looked up in a single request")
+	apiFlags.Uint32Var(&config.MaxBulkExportRelationshipsLimit, "max-bulk-export-relationships-limit", 10_000, "maximum number of relationships that can be exported in a single request")
+
+	datastoreFlags := nfs.FlagSet("Datastore")
 	// Flags for the datastore
-	if err := datastore.RegisterDatastoreFlags(cmd, &config.DatastoreConfig); err != nil {
+	if err := datastore.RegisterDatastoreFlags(cmd, datastoreFlags, &config.DatastoreConfig); err != nil {
 		return err
 	}
+	// TODO: should this be under datastore or api?
+	datastoreFlags.DurationVar(&config.SchemaWatchHeartbeat, "datastore-schema-watch-heartbeat", 1*time.Second, "heartbeat time on the schema watch in the datastore (if supported). 0 means to default to the datastore's minimum.")
 
-	// Flags for configuring the API usage of the datastore
-	cmd.Flags().Uint64Var(&config.MaxDatastoreReadPageSize, "max-datastore-read-page-size", 1_000, "limit on the maximum page size that we will load into memory from the datastore at one time")
-
+	namespaceCacheFlags := nfs.FlagSet("Namespace Cache")
 	// Flags for the namespace cache
-	cmd.Flags().Duration("ns-cache-expiration", 1*time.Minute, "amount of time a namespace entry should remain cached")
-	if err := cmd.Flags().MarkHidden("ns-cache-expiration"); err != nil {
+	namespaceCacheFlags.Duration("ns-cache-expiration", 1*time.Minute, "amount of time a namespace entry should remain cached")
```

Comment:
Maybe we should call this `Schema Cache` and make new flags i.e. `--schema-cache-expiration`? (we would need aliases to the old names so they still work). We don't really call anything "namespace" in the public API anymore so it seems out of place.

Reply:
Yeah, renaming flags is one of the things that Jimmy was doing in #2022, with the idea that they'd be aliases for each other. I'm planning on doing some of that work in a subsequent pass.